src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt

```diff
@@ -30,26 +31,48 @@ class ResourceServerConfiguration {
       .roles(role)
   }
 
+  @Order(1)
   @Bean
-  fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
+  fun permittedFilterChain(http: HttpSecurity): SecurityFilterChain {
     return http
       .csrf { it.disable() } // csrf not needed on a rest api
-      .authorizeHttpRequests { it ->
-        it.requestMatchers(
-          "/favicon.ico",
-          "/health/**",
-          "/info",
-          "/v3/api-docs/**",
-          "/v3/api-docs.yaml",
-          "/swagger-ui/**",
-          "/swagger-ui.html",
-          "/metrics/**",
-        )
-          .permitAll()
-        it.requestMatchers("/prometheus").hasRole("ACTUATOR").and().httpBasic()
+      .securityMatcher(
+        "/favicon.ico",
+        "/health/**",
+        "/info",
+        "/v3/api-docs/**",
+        "/v3/api-docs.yaml",
+        "/swagger-ui/**",
+        "/swagger-ui.html",
+        "/metrics/**",
+      )
+      .authorizeHttpRequests {
+        it.anyRequest().permitAll()
+      }
+      .build()
+  }
+
+  @Order(2)
+  @Bean
+  fun prometheusFilterChain(http: HttpSecurity): SecurityFilterChain {
+    return http
+      .csrf { it.disable() }
+      .securityMatcher("/prometheus")
+      .authorizeHttpRequests {
+        it.anyRequest().hasRole("ACTUATOR")
```

Comment:
Whilst we're here, we hard code the role on this line, but the role is configurable for the user itself - I wonder if we should either configure this one, or hard code the other role?

Reply:
Yeah wasn't sure what that was about - @oskarwilliams  what was the intention?

Reply:
I'm looking to fiddle with the actuator stuff a bit, but will do that in a separate change. Will leave this until then